spec/ace.md

```diff
@@ -49,6 +51,14 @@ Volumes that are specified in the Pod Manifest are mounted into each of the apps
 For example, say that the worker-backup and reduce-worker both have a `mountPoint` named "work".
 In this case, the executor will bind mount the host's `/opt/tenant1/work` directory into the `path` of each of the matching "work" `mountPoint`s of the two app filesystems.
 
+If the target `path` does not exist in the rendered filesystem, it SHOULD be created, including any missing parent directories. If the target `path` is a non-empty directory, its contents SHOULD be discarded (e.g. obscured by the bind mount). If the target `path` refers to a file, ACE SHOULD remove that file and create a directory in its place.
+
+Mount point `path` directories that ACE creates SHOULD be owned by UID 0 and GID 0, and have access mode `0755` (`rwxr-xr-xr-x`). ACE implementation MAY provide a method for administrator to specify different permissions on a per-pod basis.
+
+ACE MUST NOT create any paths in the host file system, and is REQUIRED to consider missing volume source paths as an error. If an underlying operating system supports it ACE MAY implement single-file volumes.
```

Comment:
If ACE can be tricked into actively altering host filesystem, this may lead to security vulnerabilities. It seems to be common sense for implementation, but the security concerns are grave enough to mention it in the spec. A similar issue would be requiring ACE to refuse to use symbolic links as sources (or even to refuse to use any path that has a symbolic link in it as a volume source); would this be a good place to add that part, or should I go to pod.md where volume is described?

If you are concerned about the spec being too strict, I'm fine with downgrading this requirement to a SHOULD level, with an addition that if the implementation supports altering host filesystem, it should be possible to turn it off.